lib/view_model/lightning_view_model.dart

```diff
@@ -0,0 +1,103 @@
+import 'dart:async';
+import 'package:breez_sdk/breez_sdk.dart';
+import 'package:breez_sdk/bridge_generated.dart' as BZG;
+import 'package:cake_wallet/entities/calculate_fiat_amount_raw.dart';
+import 'package:cake_wallet/entities/fiat_currency.dart';
+import 'package:cake_wallet/lightning/lightning.dart';
+import 'package:cake_wallet/store/dashboard/fiat_conversion_store.dart';
+import 'package:cake_wallet/store/settings_store.dart';
+import 'package:cw_core/crypto_currency.dart';
+import 'package:mobx/mobx.dart';
+
+part 'lightning_view_model.g.dart';
+
+class LightningViewModel = LightningViewModelBase with _$LightningViewModel;
+
+abstract class LightningViewModelBase with Store {
+  LightningViewModelBase({
+    required this.settingsStore,
+    required this.fiatConversionStore,
+  }) {}
+
+  final SettingsStore settingsStore;
+  final FiatConversionStore fiatConversionStore;
+  
+  @observable
+  bool loading = false;
+
+  @action
+  void setLoading(bool value) {
+    loading = value;
+  }
+
+  FiatCurrency get fiat => settingsStore.fiatCurrency;
+
+  String formattedFiatAmount(int sats) {
+    String amount = calculateFiatAmountRaw(
+      cryptoAmount: lightning!.formatterLightningAmountToDouble(amount: sats),
+      price: fiatConversionStore.prices[CryptoCurrency.btcln],
+    );
+    return amount;
+  }
+
+  Future<List<String>> receiveOnchain() async {
+    final sdk = await BreezSDK();
+
+    BZG.ReceiveOnchainRequest req = const BZG.ReceiveOnchainRequest();
+    BZG.SwapInfo swapInfo = await sdk.receiveOnchain(req: req);
+    print("Minimum amount allowed to deposit in sats: ${swapInfo.minAllowedDeposit}");
+    print("Maximum amount allowed to deposit in sats: ${swapInfo.maxAllowedDeposit}");
+    int fee = swapInfo.channelOpeningFees?.minMsat ?? 2000;
+    fee = fee ~/ 1000;
+    return [
+      swapInfo.bitcoinAddress,
+      swapInfo.minAllowedDeposit.toString(),
+      swapInfo.maxAllowedDeposit.toString(),
+      fee.toString(),
```

Comment:
**Enhancement:**
encapsulate them in an object and return this object instead.
for all functions not just this one